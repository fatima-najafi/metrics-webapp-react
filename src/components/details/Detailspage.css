.parent {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  width: 100%;
}

.items {
  width: 100%;
  height: 484px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: gray;
  margin-bottom: 0;
  text-align: center;
}

.image {
  border-radius: 10px;
  margin-top: 60px;
  padding: 1px;
  height: 100px;
  width: 98%;
  border-top: none;
}

h1 {
  margin-top: 70px;
  font-size: 15px;
}

p {
  margin-top: 50px;
}

.table {
  width: 100%;
  height: 100%;
  background-color: rgb(235, 233, 233);
  border-radius: 0 0 10px 10px;
  font-size: 14px;
}

tr {
  height: 35px;
  background: #80808091;
}

tr:nth-child(even) {
  background-color: gray;
  height: 35px;
}

/* for desctop version */
@media only screen and (min-width: 768px) {
  .parent {
    display: flex;
    justify-content: center;
    align-content: center;
    width: 100%;
    margin-top: 0;
  }

  .items {
    width: 800px;
    height: 570px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 10px;
    margin-top: 20px;
    margin-bottom: 0;
    text-align: center;
  }

  .image {
    height: 150px;
    width: 200px;
    border-radius: 10px;
    border: 5px solid rgb(255, 255, 255);
    margin-top: 20px;
    padding: 1px;
  }

  h1 {
    margin-top: -10px;
    font-size: 20px;
  }

  .table {
    border: 1px solid rgb(255, 255, 255);
    width: 100%;
    height: 100%;
    border-radius: 0 0 10px 10px;
    font-size: 20px;
  }
}
