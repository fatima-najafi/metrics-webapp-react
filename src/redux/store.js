import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import companiesReducer from './companySlice';

const store = configureStore({
  reducer: {
    companies: companiesReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
