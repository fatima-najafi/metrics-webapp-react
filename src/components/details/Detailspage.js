import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getCompanyDetails } from '../../redux/companySlice';
import './Detailspage.css';

const Detailspage = () => {
  const { companyDetails } = useSelector((state) => state.companies);
  const dispatch = useDispatch();
  const { smbl } = useParams();

  useEffect(() => {
    dispatch(getCompanyDetails(smbl));
  }, [dispatch, smbl]);

  return (
    <div className="parent" data-testid="detailspage">
      <div className="items">
        <div
          className="fix-image"
          style={{ display: 'flex', flexDirection: 'row', gap: '30px' }}
        >
          <img className="image" src={companyDetails.image} alt="company" />
          <div style={{ display: 'flex', flexDirection: 'column' }}>
            <p>
              (
              {companyDetails.symbol}
              )
            </p>
            <h1>{companyDetails.companyName}</h1>
          </div>
        </div>
        <table className="table">
          <tbody className="content">
            <tr>
              <th scope="col">CEO</th>
              <td>{companyDetails.ceo}</td>
            </tr>
            <tr>
              <th scope="col">Sector</th>
              <td>{companyDetails.sector}</td>
            </tr>
            <tr>
              <th scope="col">Price</th>
              <td>{companyDetails.price}</td>
            </tr>
            <tr>
              <th scope="col">Currency</th>
              <td>{companyDetails.currency}</td>
            </tr>
            <tr>
              <th scope="col">Full Time Employees</th>
              <td>{companyDetails.fullTimeEmployees}</td>
            </tr>
            <tr>
              <th scope="col">Country</th>
              <td>{companyDetails.country}</td>
            </tr>
            <tr>
              <th scope="col">City</th>
              <td>{companyDetails.city}</td>
            </tr>
            <tr>
              <th scope="col">Phone</th>
              <td>{companyDetails.phone}</td>
            </tr>
            <tr>
              <th scope="col">Address</th>
              <td>{companyDetails.address}</td>
            </tr>
            <tr>
              <th scope="col">Website</th>
              <td>{companyDetails.website}</td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default Detailspage;
