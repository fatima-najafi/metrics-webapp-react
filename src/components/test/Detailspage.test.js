import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import store from '../../redux/store';
import Detailspage from '../details/Detailspage';

it('renders the Detailspage component', () => {
  render(
    <Provider store={store}>
      <Detailspage />
    </Provider>,
  );
  const detailsPage = screen.getByTestId('detailspage');
  expect(detailsPage).toBeInTheDocument();
  const companyName = screen.getByText('Sector');
  expect(companyName).toBeInTheDocument();
});
