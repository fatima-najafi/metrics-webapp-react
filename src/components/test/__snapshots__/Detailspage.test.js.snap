// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check any changes to the component 1`] = `
<div
  className="parent"
  data-testid="detailspage"
>
  <div
    className="items"
  >
    <img
      alt="company"
      className="image"
    />
    <p>
      (
      )
    </p>
    <h1 />
    <table
      className="table"
    >
      <tbody>
        <tr>
          <th
            scope="col"
          >
            CEO
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Sector
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Price
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Currency
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Full Time Employees
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Country
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            City
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Phone
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Address
          </th>
          <td />
        </tr>
        <tr>
          <th
            scope="col"
          >
            Website
          </th>
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
